File: app/src/test/java/com/an/github/api/GithubApiServiceTest.java
package com.an.github.api;

import com.an.github.data.remote.api.GithubApiService;
import com.an.github.data.remote.model.GithubApiResponse;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

import retrofit2.Response;

public class GithubApiServiceTest extends ApiAbstract<GithubApiService> {

    private GithubApiService service;

    @Before
    public void initService() {
        this.service = createService(GithubApiService.class);
    }

    @Test
    public void fetchPostsTest() throws IOException {
        enqueueResponse("test_repositories.json");
        Response<GithubApiResponse> response = service.fetchRepositories("stars", "desc", 1l).blockingFirst();
        Assert.assertEquals(true, response.isSuccessful());

        GithubApiResponse apiResponse = response.body();
        Assert.assertEquals(new Long(806201), apiResponse.getTotalCount());
        Assert.assertEquals(10, apiResponse.getItems().size());
    }
}

File: app/src/test/java/com/an/github/api/GithubApiServiceTest.kt
package com.an.github.api

import com.an.github.data.remote.api.GithubApiService

import java.io.IOException

import retrofit2.Response

class GithubApiServiceTest : ApiAbstract<GithubApiService>() {

    private lateinit var service: GithubApiService

//    @Before
//    fun initService() {
//        this.service = createService(GithubApiService::class.java)
//    }
//
//    @Test
//    @Throws(IOException::class)
//    fun fetchPostsTest() {
//        enqueueResponse("test_repositories.json")
//        val response = service.fetchRepositories("stars", "desc", 1L).blockingFirst()
//        Assert.assertEquals(true, response.isSuccessful)
//
//        val apiResponse = response.body()
//        Assert.assertEquals(806201L, apiResponse?.totalCount)
//        Assert.assertEquals(10, apiResponse?.items?.size)
//    }
}
